lib/Remote/RemoteWebDriver.php

```diff
@@ -110,23 +120,45 @@ public static function create(
             $executor->setRequestTimeout($request_timeout_in_ms);
         }
 
+        // W3C
+        $parameters = [
+            'capabilities' => [
+                'firstMatch' => [$desired_capabilities->toArray()],
+            ],
+        ];
+
+        // Legacy protocol
+        if (null !== $required_capabilities && $required_capabilities_array = $required_capabilities->toArray()) {
+            $parameters['capabilities']['alwaysMatch'] = $required_capabilities_array;
+        }
+
         if ($required_capabilities !== null) {
             // TODO: Selenium (as of v3.0.1) does accept requiredCapabilities only as a property of desiredCapabilities.
-            // This will probably change in future with the W3C WebDriver spec, but is the only way how to pass these
-            // values now.
+            // This has changed with the W3C WebDriver spec, but is the only way how to pass these
+            // values with the legacy protocol.
             $desired_capabilities->setCapability('requiredCapabilities', $required_capabilities->toArray());
         }
 
+        $parameters = [
```

Comment:
this should not override the variable entirely, put assign `$parameters['desiredCapabilities']` only